autode/wrappers/QChem.py

```diff
@@ -0,0 +1,502 @@
+        return False
+
+    def get_final_atoms(self, calc) -> List[Atom]:
+
+        if isinstance(calc.input.keywords, kws.HessianKeywords):
+            logger.warning('Hessian calculation performed - no change to '
+                           'geometry')
+            return calc.molecule.atoms
+
+        atoms = []
+
+        for i, line in enumerate(calc.output.file_lines):
+
+            if 'Coordinates (Angstroms)' not in line:
+                continue
+
+            """e.g.
+            
+                               Coordinates (Angstroms)
+             ATOM                X               Y               Z
+              1  O         0.0003489977   -0.1403224128    0.0000000000
+              2  H        -0.7524338562    0.4527672831    0.0000000000
+              3  H         0.7551329498    0.4500625364    0.0000000000
+            Point Group: cs    Number of degrees of freedom:     3
+            """
+
+            start_idx = i+2
+            end_idx = start_idx + calc.molecule.n_atoms
+
+            atoms = [Atom(_l.split()[1], *_l.split()[2:5])
+                     for _l in calc.output.file_lines[start_idx:end_idx]]
+
+        return atoms
+
+    def get_atomic_charges(self, calc) -> List:
+        raise NotImplementedError
+
+    def get_energy(self, calc) -> float:
+        """Get the total electronic energy from the calculation"""
+
+        for line in reversed(calc.output.file_lines):
+
+            if 'Total energy' in line:
+                try:
+                    return float(line.split()[-1])
+
+                except (TypeError, ValueError, IndexError):
+                    break
+
+        raise CouldNotGetProperty('energy')
+
+    def get_gradients(self, calc) -> np.ndarray:
+        """Gradient of the potential energy"""
+
+        grad = []
+
+        for i, line in enumerate(calc.output.file_lines):
+
+            if 'Cartesian Gradient' not in line:
+                continue
+
+            """e.g.
+            
+                        Cartesian Gradient (au)
+             ATOM              X           Y           Z
+            1  O           0.000005   -0.000002    0.000000
+            2  H           0.000017    0.000001    0.000000
+            3  H          -0.000021    0.000001    0.000000
+            """
+
+            start_idx = i+2
+            end_idx = start_idx+calc.molecule.n_atoms
+
+            grad = [[float(val) for val in _l.split()[2:]]
+                    for _l in calc.output.file_lines[start_idx:end_idx]]
+
+        # Convert from Ha a0^-1 to Ha A-1
+        return np.array(grad) / Constants.a0_to_ang
+
+    def get_hessian(self, calc) -> np.ndarray:
+        """Extract the mass-weighted non projected Hessian matrix
+        NOTE: Required $rem vibman_print 4 $end in the input"""
+        expected_shape = (3 * calc.molecule.n_atoms, 3 * calc.molecule.n_atoms)
+
+        hess = self._extract_mass_weighted_hessian(calc)
+        if hess.shape != expected_shape:
```

Comment:
I thijnk this necessarily has the correct shape from the logic in the other function?

Reply:
unless it isn't found at all?

Reply:
Agree - the exception for a not-found Hessian should be raised in `_extract_mass_weighted_hessian`